bin/aws-configure

```diff
@@ -39,72 +31,55 @@ To add a new profile, you MUST specify following environment variables:
 
 You can optionally specify '<PROFILE>_AWS_DEFAULT_REGION' to determine the default region for this profile.
 
-If no PROFILE is specified, ${0##*/} will defer to the value specified in AWS_PROFILE.
-
 SUBCOMMANDS:
     is-configured PROFILE     Returns '0' if the profile is configured. Otherwise, it returns '1'.
 DOC
 }
 
 function setup_profile() {
-  profile="$1"
-  profile_env="$(echo "${profile^^}" | tr '-' '_')"
+  profile=$1
```

Comment:
I should've made this comment last time, but there's a shorthand for the whole idiom of exiting if an expected parameter is null or unset. You can replace the entirety of `setup_profile` with:
```bash
  profile=${1:?No profile specified. You must specify an AWS profile to configure.}
```
This would generate output similar to this (and exit):
```
/tmp/test.sh: line 3: 1: No profile specified. You must specify an AWS profile to configure.
```
So if you prefer having more control over the error message the existing approach is totally fine.

I always have trouble remembering the different parameter expansion options, so I find [this table](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) invaluable.